File: address-validation-us-3-dot-net-examples/ValidateCityStateZipSoapSdkExample.cs
using address_validation_us_3_dot_net.SOAP;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static address_validation_us_3_dot_net.REST.GetBestMatchesClient;

namespace address_validation_us_3_dot_net_examples
{
    internal static class ValidateCityStateZipSoapSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        { 
            Console.WriteLine("\r\n---------------------------------------------------------");
            Console.WriteLine("Address Validation US 3 - ValidateCityStateZip - SOAP SDK");
            Console.WriteLine("---------------------------------------------------------");

            Console.WriteLine("\r\n* Input *\r\n");

            Console.WriteLine($"City       : Santa Barbara");
            Console.WriteLine($"State      : CA");
            Console.WriteLine($"ZIP Code   : 93101");
            Console.WriteLine($"Is Live    : {IsLive.ToString()}");
            Console.WriteLine($"License Key: {LicenseKey}");


            CityStateZipValidation cityStateZipValidation = new(IsLive);
            AV3Service.CityStateZipResponse response = cityStateZipValidation.ValidateCityStateZip(
                "Santa Barbara",
                "CA",
                "93101",
                LicenseKey
            );

            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Validation *\r\n");

                Console.WriteLine($"City                          : {response.CityStateZip.City}");
                Console.WriteLine($"State                         : {response.CityStateZip.State}");
                Console.WriteLine($"ZIP Code                      : {response.CityStateZip.Zip}");
                Console.WriteLine($"General Delivery Service      : {response.CityStateZip.GeneralDeliveryService}");
                Console.WriteLine($"Street Service                : {response.CityStateZip.StreetService}");
                Console.WriteLine($"PO Box Service                : {response.CityStateZip.POBoxService}");
                Console.WriteLine($"PO Box Range Low              : {response.CityStateZip.POBoxRangeLow}");
                Console.WriteLine($"PO Box Range High             : {response.CityStateZip.POBoxRangeHigh}");
                Console.WriteLine($"Rural Route / Highway Contract: {response.CityStateZip.RRHCService}");
                Console.WriteLine($"Urbanization Service          : {response.CityStateZip.UrbanizationService}");
                Console.WriteLine($"Is Unique ZIP Code            : {response.CityStateZip.IsUniqueZipCode}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");

                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: address-validation-us-3-python-examples/validate_city_state_zip_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-us-3-python/SOAP"))

from validate_city_state_zip_soap import ValidateCityStateZipValidation

def validate_city_state_zip_soap_sdk_go(license, is_live_key):
    print("\n" + "-" * 57)
    print("Address Validation US 3 - ValidateCityStateZip - SOAP SDK")
    print("-" * 57)

    print("\n* Input *\n")
    print(f"City       : Santa Barbara")
    print(f"State      : CA")
    print(f"ZIP Code   : 93101")
    print(f"Is Live    : {is_live_key}")
    print(f"License Key: {license}")

    try:
        service = ValidateCityStateZipValidation(license, is_live=is_live_key, timeout_ms=10000)
        response = service.validate_city_state_zip(
            city="Santa Barbara",
            state="CA",
            zip="93101"
        )

        if not hasattr(response, 'Error'):
            print("\n* Validation *\n")
            city_state_zip = response.CityStateZip
            print(f"City                  : {city_state_zip.City}")
            print(f"State                 : {city_state_zip.State}")
            print(f"ZIP                   : {city_state_zip.Zip}")             
            print(f"GeneralDeliveryService: {city_state_zip.GeneralDeliveryService}")
            print(f"POBoxService          : {city_state_zip.POBoxService}")
            print(f"StreetService         : {city_state_zip.StreetService}")
            print(f"RRHCService           : {city_state_zip.RRHCService}")
            print(f"UrbanizationService   : {city_state_zip.UrbanizationService}")
            print(f"POBoxRangeLow         : {city_state_zip.POBoxRangeLow}")
            print(f"POBoxRangeHigh        : {city_state_zip.POBoxRangeHigh}")
            print(f"IsUniqueZipCode       : {city_state_zip.IsUniqueZipCode}")
        else:
            print("\n* Error *\n")
            err = response.Error
            print(f"Error Type     : {err.Type}")
            print(f"Error Type Code: {err.TypeCode}")
            print(f"Error Desc     : {err.Desc}")
            print(f"Error Desc Code: {err.DescCode}")
    except RuntimeError as e:
        print("Error calling service:", e)
